src/modules/powerrename/lib/PowerRenameRegEx.cpp

```diff
@@ -201,9 +233,17 @@ HRESULT CPowerRenameRegEx::Replace(_In_ PCWSTR source, _Outptr_ PWSTR* result)
         try
         {
             // TODO: creating the regex could be costly.  May want to cache this.
+            wchar_t newReplaceTerm[MAX_PATH] = { 0 };
+            bool fileTimeErrorOccurred = false;
+            if (m_useFileTime)
+            {
+                if (FAILED(GetDatedFileName(newReplaceTerm, ARRAYSIZE(newReplaceTerm), m_replaceTerm, m_fileTime)))
+                    fileTimeErrorOccurred = true;
+            }
+
             std::wstring sourceToUse(source);
             std::wstring searchTerm(m_searchTerm);
-            std::wstring replaceTerm(m_replaceTerm ? wstring(m_replaceTerm) : wstring(L""));
+            std::wstring replaceTerm((m_useFileTime && !fileTimeErrorOccurred) ? wstring(newReplaceTerm) : (m_replaceTerm ? wstring(m_replaceTerm) : wstring(L"")));
```

Comment:
Please split this in more maintainable format, we want to keep the code as clean and readable as possible.